server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java

```diff
@@ -0,0 +1,213 @@
+package org.infinispan.server.resp;
+
+import static org.infinispan.server.resp.test.RespTestingUtil.createClient;
+import static org.infinispan.server.resp.test.RespTestingUtil.killClient;
+import static org.infinispan.server.resp.test.RespTestingUtil.killServer;
+import static org.infinispan.server.resp.test.RespTestingUtil.startServer;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import io.lettuce.core.KeyValue;
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.api.StatefulRedisConnection;
+import io.lettuce.core.api.sync.RedisCommands;
+import io.lettuce.core.pubsub.RedisPubSubAdapter;
+import io.lettuce.core.pubsub.api.sync.RedisPubSubCommands;
+
+/**
+ * Base class for single node tests.
+ *
+ * @author William Burns
+ * @since 14.0
+ */
+@Test(groups = "functional", testName = "server.resp.RespSingleNodeTest")
+public class RespSingleNodeTest extends SingleCacheManagerTest {
+   protected RedisClient client;
+   protected RespServer server;
+   protected StatefulRedisConnection<String, String> redisConnection;
+   protected static final int timeout = 60;
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() {
+      cacheManager = createTestCacheManager();
+      server = startServer(cacheManager);
+      client = createClient(30000, server.getPort());
+      redisConnection = client.connect();
+      cache = cacheManager.getCache(server.getConfiguration().defaultCacheName());
+      return cacheManager;
+   }
+
+   protected EmbeddedCacheManager createTestCacheManager() {
+      return TestCacheManagerFactory.createCacheManager(true);
+   }
+
+   @AfterClass(alwaysRun = true)
+   @Override
+   protected void destroyAfterClass() {
+      super.destroyAfterClass();
+      log.debug("Test finished, close resp server");
+      killClient(client);
+      killServer(server);
+   }
+
+   public void testSetMGet() {
+      RedisCommands<String, String> redis = redisConnection.sync();
+      redis.set("k1", "v1");
```

Comment:
does RESP support POJO?

Reply:
It is normally for string keys and values. TBH I am not sure how arbitrary pojos work. I know the value can be a byte[] though.

Reply:
what about numbers (int/long/double)? 
it would be nice the extend the test to include `byte[]` too.